src/integrationTest/java/com/mongodb/hibernate/query/select/SimpleSelectQueryIntegrationTests.java

```diff
@@ -0,0 +1,186 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.query.select;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.mongodb.hibernate.junit.MongoExtension;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.util.List;
+import java.util.function.Consumer;
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.query.SelectionQuery;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.jspecify.annotations.Nullable;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@SessionFactory(exportSchema = false)
+@DomainModel(annotatedClasses = SimpleSelectQueryIntegrationTests.Contact.class)
+@ExtendWith(MongoExtension.class)
+class SimpleSelectQueryIntegrationTests {
+
+    @BeforeEach
+    void beforeEach(SessionFactoryScope scope) {
+        scope.inTransaction(session -> List.of(
+                        new Contact(1, "Bob", 18, Country.USA),
+                        new Contact(2, "Mary", 35, Country.CANADA),
+                        new Contact(3, "Dylan", 7, Country.CANADA),
+                        new Contact(4, "Lucy", 78, Country.CANADA),
+                        new Contact(5, "John", 25, Country.USA))
+                .forEach(session::persist));
+    }
+
+    @Test
+    void testComparisonByEQ(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session,
+                "from Contact where country = :country",
+                q -> q.setParameter("country", Country.USA.name()),
+                List.of(1, 5)));
+    }
+
+    @Test
+    void testComparisonByNE(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session,
+                "from Contact where country != ?1",
+                q -> q.setParameter(1, Country.USA.name()),
+                List.of(2, 3, 4)));
+    }
+
+    @Test
+    void testComparisonByLT(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session, "from Contact where age < :age", q -> q.setParameter("age", 35), List.of(1, 3, 5)));
+    }
+
+    @Test
+    void testComparisonByLTE(SessionFactoryScope scope) {
```

Comment:
(consistency) Acronyms should use init-upper case, as for example you did with `SqlAstTranslatorFactory`

Reply:
changed to `testComparisonByLte`, together with other similar changes.